lib/cache.js

```diff
@@ -202,8 +202,8 @@ function updateQueries(curr, { serviceName, method, item }, { idField, updatedAt
 
   for (const item of items) {
     const itemId = idField(item)
-    const queries = { ...getIn(curr, ['queries', serviceName]) }
-    const index = { ...getIn(curr, ['index', serviceName, itemId]) }
+    const queries = { ...getIn(next, ['queries', serviceName]) }
+    const index = { ...getIn(next, ['index', serviceName, itemId]) }
```

Comment:
This is the fix. I renamed curr->next in other places for consistency / to avoid similar issue if code is updated. The idea is `curr` is the current state at the beginning of the mutation, `next` is the mutated version that we evolve and return. Reading from `curr` here inside a loop was a mistake because we kept seeing the original state of the query items array, not the one we've been updating via the loop.